File: add-ons/lookups/payfone-tcpa-compliance/payfone-tcpa-compliance.5.x.cs
// Download the C# helper library from twilio.com/docs/libraries/csharp
using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Lookups.V1;
using Twilio.Types;

namespace Example
{
    class MainClass
    {
        public static void Main(string[] args)
        {
            // Get your Account SID and Auth Token from twilio.com/console
            const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
            const string authToken = "your_auth_token";

            TwilioClient.Init(accountSid, authToken);

            var addOns = new List<string> { "payfone_tcpa_compliance" };
            var addOnsData = new Dictionary<string, object>
            {
                { "payfone_tcpa_compliance.RightPartyContactedDate", "20160101" }
            };

            var phoneNumberFetcher =
                PhoneNumberResource.Fetcher(new PhoneNumber("+16502530000"));
            phoneNumberFetcher.addOns = addOns;
            phoneNumberFetcher.addOnsData = addOnsData;

            var number = phoneNumberFetcher.Fetch();
            Console.WriteLine(number.addOns);
        }
    }
}

File: add-ons/lookups/payfone-tcpa-compliance/payfone-tcpa-compliance.5.x.js
// Get your Account SID and Auth Token from twilio.com/console
// To set up environmental variables, see http://twil.io/secure
const accountSid = process.env.TWILIO_ACCOUNT_SID;
const authToken = process.env.TWILIO_AUTH_TOKEN;

// Download the Node.js helper library from twilio.com/docs/libraries/node
const client = require('twilio')(accountSid, authToken);

client.lookups.v1
  .phoneNumbers('+16502530000')
  .fetch({
    addOns: 'payfone_tcpa_compliance',
    addOnsData: {
      'payfone_tcpa_compliance.RightPartyContactedDate': '20160101',
    },
  })
  .then(number => console.log(number.addOns));
